tools/marvin/marvin/lib/base.py

```diff
@@ -3876,7 +3876,12 @@ def add(cls, apiclient, services, physicalnetworkid,
         else:
             cmd.transportzoneuuid = services['transportZoneUuid']
 
-        return NiciraNvp(apiclient.addNiciraNvpDevice(cmd).__dict__)
+        if l2gatewayserviceuuid:
+            cmd.l2gatewayserviceuuid = l2gatewayserviceuuid
+        else:
+            cmd.l2gatewayserviceuuid = services['l2gatewayserviceuuid']
```

Comment:
Never mind this one, I've done it myself and I will ask you to pull in my changes to your PR asap.